docs/pages/repo/docs/core-concepts/caching/file-inputs.mdx

```diff
@@ -0,0 +1,116 @@
+---
+title: File Inputs
+description: How to configure Turborepo file inputs
+---
+
+import Callout from "../../../../../components/Callout";
+
+# File Inputs
+
+By default, a workspace is considered to have been updated when any of the files in that workspace have changed. However, not every file change inside of the entire repo should result in rerunning a task. Turborepo gives you the tools to make sure that you can define every file that matters.
+
+## Global Files
+
+Almost every repository has some configuration that appears at the root of the repository. These files have the potential to impact the behavior of child workspaces.
+
+### `package.json` and `turbo.json`
+
+By default, Turborepo includes the root `package.json`, any lockfile that may be present, and the root `turbo.json` file into the global hash. You do not need to specify them separately.
```

Comment:
We don't include the lockfile anymore. We include a hash of resolved external root deps, but lockfile changes should now only invalidate for workspaces that depend on whatever changed.

Reply:
There was an earlier iteration that specified that we grab relevant bits from the lockfile but I deleted it because it was way too much detail for a section on file inputs.